src/Compilers/Core/Portable/Operations/ControlFlowGraphBuilder.cs

```diff
@@ -1998,18 +2003,53 @@ public override IOperation VisitInvocation(IInvocationOperation operation, int?
             return new InvocationExpression(operation.TargetMethod, visitedInstance, operation.IsVirtual, visitedArguments, semanticModel: null, operation.Syntax, operation.Type, operation.ConstantValue, operation.IsImplicit);
         }
 
+        private int? _currentInitializerCaptureId = null;
+
         public override IOperation VisitObjectCreation(IObjectCreationOperation operation, int? captureIdForResult)
         {
             ImmutableArray<IArgumentOperation> visitedArgs = VisitArguments(operation.Arguments);
 
             // Initializer is removed from the tree and turned into a series of statements that assign to the created instance
-            var objectCreation = new ObjectCreationExpression(operation.Constructor, initializer: null, visitedArgs, semanticModel: null, operation.Syntax, operation.Type, operation.ConstantValue, operation.IsImplicit);
+            IOperation objectCreation = new ObjectCreationExpression(operation.Constructor, initializer: null, visitedArgs, semanticModel: null, operation.Syntax, operation.Type, operation.ConstantValue, operation.IsImplicit);
 
-            // PROTOTYPE(dataflow): For a non-null initializer, we'll need to assign the created object to a flow reference and rewrite all initializers to assign to/call functions on it, then return return the flow reference
+            // PROTOTYPE(dataflow): Handle collection initializers
+            if (operation.Initializer != null)
+            {
+                SpillEvalStack();
+
+                int? previousInitializerId = _currentInitializerCaptureId;
+
+                int currentInitializerCaptureId = _availableCaptureId++;
+                _currentInitializerCaptureId = currentInitializerCaptureId;
+
+                AddStatement(new FlowCapture(currentInitializerCaptureId, objectCreation.Syntax, objectCreation));
+
+                foreach (IOperation initializer in operation.Initializer.Initializers)
+                {
+                    AddStatement(Visit(initializer));
+                }
+
+                objectCreation = new FlowCaptureReference(currentInitializerCaptureId, objectCreation.Syntax, objectCreation.Type, objectCreation.ConstantValue);
+
+                _currentInitializerCaptureId = previousInitializerId;
+            }
 
             return objectCreation;
         }
 
+        public override IOperation VisitInstanceReference(IInstanceReferenceOperation operation, int? captureIdForResult)
+        {
+            if (operation.ReferenceKind == InstanceReferenceKind.Initializer)
```

Comment:
>if (operation.ReferenceKind == InstanceReferenceKind.Initializer) [](start = 12, length = 65)

Can we make any adjustments to CanBeInControlFlowGraph helper? For example, that we don't expect to run into this kind of IInstanceReferenceOperation.